src/components/BalanceService.tsx

```diff
@@ -1,4 +1,5 @@
 import MemoryService from "../core/Memory";
+import { makeAutoObservable } from "mobx"
 
 class BalanceService {
     balances: { date: Date, value: number }[];
```

Comment:
It should contain method getBalances() that should be used to access balances variable.
Then mobx can track its changes.